### PR TITLE
[grafana] Add Red Hat OpenShift support, change image to grafana-enterprise, allow…

### PR DESCRIPTION
**Pull request by patrickabrennan**
fields for the Grafana Enterprise license, turn PVC on so if the Grafana Pod restarts data will not be lost. Also by using the grafana-enterprise image one has all the features of community Grafana image, but easily all customers to enable the enterprise version.

### CODE REVIEW

File: charts/grafana/values.yaml
Comment (vlad-diachenko) [inline charts/grafana/values.yaml:440]: this secret must be manually created outside the helm chart. according to this:
```
## Additional grafana server secret mounts
# Defines additional mounts with secrets. Secrets must be manually created in the namespace.
```
So, not enterprise users won't have it and it will prevent the pod from starting up.  As long as I remember.

File: charts/grafana/values.yaml
Comment (vlad-diachenko) [inline charts/grafana/values.yaml:77]: it might be breaking change for somebody. I know that some companies do not use docker hub directly, instead of this they use company's docker repository proxy and they use only some list of allowed docker images. 
so, if they do not have this image in the white list, their k8s cluster might be not able to download the image. 
I think we need to mention it in upgrade guide: https://github.com/grafana/helm-charts/blob/main/charts/grafana/README.md#upgrading-an-existing-release-to-a-new-major-version

File: charts/grafana/values.yaml
Comment (vlad-diachenko) [inline charts/grafana/values.yaml:298]: it's better to mention it in upgrade guide

File: charts/grafana/templates/_helpers.tpl
Comment (vlad-diachenko) [inline charts/grafana/templates/_helpers.tpl:187]: for k8s version less than 1.25 we won't create pod security policy? 
user might be confused if he enabled psp in values file : `.Values.rbac.create and (eq .Values.rbac.type "psp")`
but we have not created it. 
can we add some WARNING or something like this?

File: charts/grafana/templates/_helpers.tpl
Comment (vlad-diachenko) [inline charts/grafana/templates/_helpers.tpl:184]:
```suggestion
Return if we should create a PodSecurityPolicy. Takes into account user values and supported kubernetes versions.
```

**Review by vlad-diachenko — COMMENTED** [review]
PSP and SCC config looks great. just a small nit. 
I have some doubts if we need to enable persistence and use enterprise image by default.
It's better to wait for Grafana team for the review as long as they are maintainers of this helm chart.
Also, we need to bump version of the chart   in https://github.com/grafana/helm-charts/blob/main/charts/grafana/Chart.yaml .

File: charts/grafana/values.yaml
Comment (patrickabrennan) [reply charts/grafana/values.yaml:440]: Good point. So maybe we take that out.

File: charts/grafana/values.yaml
Comment (patrickabrennan) [reply charts/grafana/values.yaml:77]: Also a very valid point.

File: charts/grafana/values.yaml
Comment (patrickabrennan) [reply charts/grafana/values.yaml:298]: While one could do that I think this would avoid some people missing to turn PVC on and thus losing data when the Pod restarts.

File: charts/grafana/templates/_helpers.tpl
Comment (patrickabrennan) [reply charts/grafana/templates/_helpers.tpl:184]: Makes total sense as psp goes away with K8s 1.25

File: charts/grafana/values.yaml
Comment (vlad-diachenko) [inline charts/grafana/values.yaml:439]: I think it's better to set empty array by default to not bring possible issues 
```suggestion
extraSecretMounts: []
```

File: charts/grafana/values.yaml
Comment (vlad-diachenko) [inline charts/grafana/values.yaml:683]: just a question: do we need to set `null` to root_url?

**Review by vlad-diachenko — COMMENTED** [review]
do we need to commit these binaries? 
* charts/grafana.original.tar
* charts/grafana.working.09232022.tar

**Review by Supra89kren — APPROVED** [review]
looks great. just bump a version of the chart and an entry to `CHANGELOG.md`

**Comment by zanhsieh** [comment]
@vlad-diachenko @patrickabrennan 
Actually I favour the approach of loki / tempo charts for documentation: use helm-doc, but it doesn't seem widely accepting by contributors. We could help the user to resolve ambiguous type of variables / stanzas with helm-docs.

**Pull request closed without merging**
